src/monitoring/monitoring.service.ts

```diff
@@ -0,0 +1,90 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { Cron, CronExpression, Timeout } from '@nestjs/schedule';
+import { AlertService } from 'src/alert/alert.service';
+import { AlertDto } from 'src/alert/dto/alert.dto';
+import { CoinMarketCapService } from 'src/coin-market-cap/coin-market-cap.service';
+import { AlertActionEnum , MailjetService} from 'src/mailjet/mailjet.service';
+
+@Injectable()
+export class MonitoringService {
+  private readonly logger = new Logger(MonitoringService.name);
+  private isInitiated: boolean = false;
+  private monitoringAlerts: AlertDto[] = [];
+  private cryptos: any[] = [];
+
+  constructor(
+    private readonly alertService: AlertService,
+    private readonly coinMarketCapService: CoinMarketCapService,
+    private readonly mailjetService: MailjetService,
+  ) {}
+
+  @Timeout(5000)
```

Comment:
**nitpick:** I don't think I understand why it's timed out after 5 seconds. Couldn't it be possible to initialize it in constructor when class is being mounted?